db/models.py

```diff
@@ -41,8 +52,77 @@ def __str__(self) -> str:
 
 class MovieSession(models.Model):
     show_time = models.DateTimeField()
-    cinema_hall = models.ForeignKey(to=CinemaHall, on_delete=models.CASCADE)
-    movie = models.ForeignKey(to=Movie, on_delete=models.CASCADE)
+    cinema_hall = models.ForeignKey(to=CinemaHall, on_delete=models.DO_NOTHING)
+    movie = models.ForeignKey(to=Movie, on_delete=models.DO_NOTHING)
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(to=User, on_delete=models.DO_NOTHING)
+
+    def __str__(self) -> str:
+        return self.created_at.strftime("%Y-%m-%d %H:%M:%S")
+
+    class Meta:
+        ordering = ["-created_at"]
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(
+        to=MovieSession,
+        on_delete=models.DO_NOTHING
+    )
+    order = models.ForeignKey(to=Order, on_delete=models.DO_NOTHING)
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    def __str__(self) -> str:
+        return f"{self.movie_session.movie.title} " \
+               f"{self.movie_session.show_time} " \
+               f"(row: {self.row}, seat: {self.seat})"
+
+    def clean(self) -> None:
+        if not 1 <= self.row <= self.movie_session.cinema_hall.rows:
+            raise ValidationError(
+                {
+                    "row":
+                        f"row number must be in available range: (1, rows): "
+                        f"(1, {self.movie_session.cinema_hall.rows})"
+                }
+            )
+        if not 1 <= self.seat <= self.movie_session.cinema_hall.seats_in_row:
+            raise ValidationError(
+                {
+                    "seat":
+                        f"seat number must be in available range: "
+                        f"(1, seats_in_row): (1, "
+                        f"{self.movie_session.cinema_hall.seats_in_row})"
+                }
+            )
+
+    def save(
+        self,
+            force_insert: bool = False,
+            force_update: bool = False,
+            using: Optional[None] = None,
+            update_fields: Optional[None] = None
+    ) -> None:
+        self.full_clean()
+        super(
+            Ticket,
+            self
+        ).save(
+            force_insert,
+            force_update,
+            using,
+            update_fields)
```

Comment:
```suggestion
        super(Ticket, self).save(
            force_insert,
            force_update,
            using,
            update_fields
        )
```
For me this will look more accurate